sql/core/src/main/scala/org/apache/spark/sql/execution/command/createDataSourceTables.scala

```diff
@@ -204,13 +194,21 @@ case class CreateDataSourceTableAsSelectCommand(
       case None => data
     }
 
+    val tableWithPath = if (table.tableType == CatalogTableType.MANAGED) {
+      table.withNewStorage(
```

Comment:
I think we should not set location Uri for managed table. As the location Uri can be deduced by database path and table name.